internal/assertion/update.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package assertion
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"time"
+
+	"github.com/fatih/color"
+	"k8s.io/klog"
+
+	"sigs.k8s.io/krew/internal/version"
+)
+
+const (
+	githubVersionURL = "https://api.github.com/repos/kubernetes-sigs/krew/releases/latest"
+
+	upgradeNotification = "You are using an old version of krew (%s). Please upgrade to the latest version (%s)"
+)
+
+// for testing
+var versionURL = githubVersionURL
+
+// CheckVersion returns a notification message to inform the user
+// about a new version of krew, or an empty string. The notification
+// will only be emitted once a day.
+func CheckVersion(basePath string) string {
+	f := filepath.Join(basePath, "last_update_check")
+	lastCheck := loadTimestamp(f)
+	tag := fetchTag(lastCheck)
+	if version.GitTag() == tag {
+		return ""
+	}
+	saveTimestamp(f)
+	return color.New(color.Bold).Sprintf(upgradeNotification, version.GitTag(), tag)
+}
+
+// fetchTag tries to return the tag name of the most recent krew
+// release. If the last check happened recently, or an error occurs
+// the hardcoded tag of the executing krew binary will be returned.
+// This effectively disables the update notification message.
+func fetchTag(lastCheck time.Time) string {
+	if time.Since(lastCheck).Hours() <= 24 {
+		klog.V(3).Info("Last check was recently, skipping update check")
+		return version.GitTag()
+	}
+
+	latestTag, err := fetchLatestTagFromGithub()
+	if err != nil || latestTag == "" {
+		klog.V(3).Infof("Could not fetch most recent tag name: %s", err)
+		return version.GitTag()
+	}
+
+	klog.V(4).Infof("Found latest tag %q", latestTag)
+	return latestTag
+}
+
+// fetchLatestTagFromGithub fetches the tag name of the latest release from github.
+func fetchLatestTagFromGithub() (string, error) {
+	response, err := http.Get(versionURL)
+	if err != nil {
+		klog.V(4).Infof("Could not GET the latest release: %s", err)
+		return "", err
+	}
+	defer response.Body.Close()
+
+	var res struct {
+		Tag string `json:"tag_name"`
+	}
+	if err := json.NewDecoder(response.Body).Decode(&res); err != nil {
+		klog.V(4).Infof("Could not parse the response from GitHub: %s", err)
+		return "", err
+	}
+	return res.Tag, nil
+}
+
+func saveTimestamp(file string) {
+	klog.V(4).Info("Saving timestamp for last version check")
+	timestamp := time.Now().Format(time.RFC1123)
```

Comment:
why not just use unix time as the persisted format?

Reply:
Do you mean seconds since epoch? I prefer a more readable format. I changed to something which still easy to manipulate.